test/smoke/src/sql/areas/notebook/notebook.test.ts

```diff
@@ -59,6 +59,33 @@ export function setup(opts: minimist.ParsedArgs) {
 			await app.workbench.sqlNotebook.waitForColorization('6', 'mtk1'); // employees
 		});
 
+		it('can enter and exit edit mode and navigate using keyboard nav', async function () {
+			const app = this.app as Application;
+			await app.workbench.sqlNotebook.newUntitledNotebook();
+			await app.workbench.sqlNotebook.addCellFromPlaceholder('Code');
+			await app.workbench.sqlNotebook.waitForPlaceholderGone();
+			assert(await app.workbench.sqlNotebook.isCodeCellInEditMode(), 'code cell should be in edit mode');
+			await app.code.dispatchKeybinding('escape');
+			assert(!(await app.workbench.sqlNotebook.isCodeCellInEditMode()), 'code cell should not be in edit mode');
+			await app.workbench.sqlNotebook.addCell('markdown');
+			assert(await app.workbench.sqlNotebook.isTextCellInEditMode(), 'text cell should be in edit mode');
+			await app.code.dispatchKeybinding('escape');
+			assert(!(await app.workbench.sqlNotebook.isTextCellInEditMode()), 'text cell should not be in edit mode');
+			await app.code.dispatchKeybinding('up');
```

Comment:
It would also be good to test : 

1. Looping from the top and bottom to the other end of the notebook
2. Testing functionality when other stuff comes up (like find widget or connection dialog)
3. That text can be entered into the editors and shows up (checking the focused state is good, but it would be best to also make absolutely sure that it's editable)
4. Using navigation keys while in edit mode doesn't navigate

I also feel a bit wary of having all these checks without checking that the cell you expect to be active/editable is actually the one found. For any of these checks if the wrong cell was made active/editable then the test would still happily pass.

Reply:
Will make a separate PR to cover this escenario